wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/environment/InternalEnvironment.java

```diff
@@ -151,6 +153,10 @@ public InternalEnvironment setWarnings(List<Warning> warnings) {
     return attributes;
   }
 
+  public DevfileImpl getDevfile() {
```

Comment:
To solve the issue with env vars propagating we could instead of putting here Devfile introduce two fields: `Component editor`, `List<Component> plugins`, and then use this as one storage instead of putting workspace config attributes + whole devfile.

Reply:
Sounds OK for me at this stage, since devfile contains a lot of info which is unused here

Reply:
Maybe another solution to the problem solved by explicitly setting the id of editors and components would be if we stored  `Map<String, Component>` for editors and plugins (or maybe separately, not sure) where the key would be the resolved FQN.

Reply:
I tried and failed.
Editor/Plugin component now is in workspace module (not k8s specific).
Moving editor/plugin configuration into environment would mean that we need make editor/plugin components do k8s/os infra specific and I'm not sure that it's a good think to do.

We also would be able to introduce editor/plugins in workspace config level BUT now they are in workspace config attributes which are propagated like in many places, and it would mean that we should propagate editor/plugins as a separate arguments to all places where it's needed...
And I'm not sure that we should invest time into such evolution. The more meaningful change would be to get rid of WorkspaceConfig at all, but it requires quite a lot of changes.

It seems that putting while Devfile into WorkspaceConfig and InternalEnvironment is the quickest option.